src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -1269,13 +1270,15 @@ MergeTreeData::LoadPartResult MergeTreeData::loadDataPart(
         throw;
     }
 #if USE_AZURE_BLOB_STORAGE
-    catch (const Azure::Core::Http::TransportException &)
+    catch (const Azure::Core::RequestFailedException & e)
```

Comment:
Updated to catch RequestFailedException azure exception as its the base exception (covers StorageException & TransportException). I think it makes sense to check for this one everywhere & retry on exception